File: Backend schronisko/Backend schronisko/Models/Zwierze.cs
namespace Backend_schronisko.Models
{
    public class Zwierze
    {
        public int Id { get; set; } // Primary Key
        public string Imie { get; set; }
        public int WiekMiesiace { get; set; }
        public string ZdjecieUrl { get; set; }

        public int GatunekId { get; set; } // Foreign Key
        public Gatunek Gatunek { get; set; } // Referencja do nawigacji
    }
}

File: Frontend-schronisko/src/types/zwierze.ts
export interface Zwierze {
  id: number
  imie: string
  gatunekId: number
  gatunek?: {
    id: number
    nazwa: string
  }
  numerEwidencyjny: string
  status: string
  numerBoksu?: string
  dataPrzyjecia: Date | string | null
  plec: string
  przyblizonaDataUrodzenia: Date | string | null
  wiekMiesiace: number
  rasa?: string
  czyZachipowany: boolean
  numerChip?: number
  umaszczenie?: string
  wielkosc?: string
  czySzczepiony: boolean
  czyKastrowanySterylizowany: boolean
  choroby?: string
  zachowanie?: string
  akceptuje?: string
  zdjecieUrl?: string
}
export type NoweZwierze = Omit<Zwierze, 'id'>;
